freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -70,23 +71,61 @@ def handle(self, event):
 
         # Get builds NVRs from the advisory attached to the message/event and
         # then get original NVR for every build
-        original_nvrs = set()
+
+        # Mapping of original build nvrs to rebuilt nvrs in advisory
+        nvrs_mapping = {}
         for product_info in event.advisory.builds.values():
             for build in product_info['builds']:
                 # Search for the first build that triggered the chain of rebuilds
                 # for every shipped NVR to get original NVR from it
                 original_nvr = self.get_published_original_nvr(build['nvr'])
                 if original_nvr is None:
                     continue
-                original_nvrs.add(original_nvr)
+                nvrs_mapping[original_nvr] = build['nvr']
 
+        original_nvrs = nvrs_mapping.keys()
         self.log_info(
             "Orignial nvrs of build in the advisory #{0} are: {1}".format(
                 event.advisory.errata_id, " ".join(original_nvrs)))
-        # Get images by nvrs and then get their digests
-        original_images_digests = self._pyxis.get_digests_by_nvrs(original_nvrs)
-        if not original_images_digests:
-            msg = f"There are no digests for NVRs: {','.join(original_nvrs)}"
+
+        # Get image manifest_list_digest for all original images, manifest_list_digest is used
+        # in pullspecs in bundle's related images
+        original_digests_by_nvr = {}
+        for nvr in original_nvrs:
+            digest = self._pyxis.get_manifest_list_digest_by_nvr(nvr)
+            if digest:
+                original_digests_by_nvr[nvr] = digest
+            else:
+                log.warning(
+                    f"Image manifest_list_digest not found for original image {nvr} in Pyxis, "
+                    "skip this image"
+                )
+
+        if not original_digests_by_nvr:
+            msg = f"None of the original images have digests in Pyxis: {','.join(original_nvrs)}"
+            log.warning(msg)
+            db_event.transition(EventState.SKIPPED, msg)
+            return []
+
+        # Convert to mapping of digest to NVR for later use
+        original_nvrs_by_digest = {v: k for k, v in original_digests_by_nvr.items()}
```

Comment:
Optional: You could just build this dictionary as part of the `for nvr in original_nvrs:` loop to avoid looping through the values again. It also makes it easier to read in my opinion.